app/uk/gov/hmrc/gvmshaulierstatus/actions/AuthorisedAction.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gvmshaulierstatus.actions
+
+import com.google.inject.Inject
+import play.api.mvc.{ActionBuilder, AnyContent, Request, Results}
+import uk.gov.hmrc.internalauth.client._
+
+import scala.concurrent.Future
+
+class AuthoriseAction @Inject() (val internalAuth: BackendAuthComponents) {
+
+  private val writePermission: Predicate.Permission = Predicate.Permission(
+    resource = Resource(
+      resourceType = ResourceType("gvms-haulier-status"),
```

Comment:
minor: I would go with the `gvms-write-haulier-status` or `gvms-haulier-status-write` type but its fine if using service name is the convention followed by others because if we have to add more **write** perms then we may be in trouble with above

Reply:
The resource being accessed in this case is `gvms-haulier-status`, here: https://github.com/hmrc/internal-auth-config/blob/main/qa/resource-definitions.yaml#L214 - the actions allowed to be performed are held in another property in that yaml file.

Reply:
for the same resourceType, we can define different `resourceLocation`s within the same service. In this case I opted for all of them (all resources, via the asterisk)

Reply:
ok, i expected more granular level perms, because if we assign WRITE to some client then they would be able to call any write endpoints which is not ideal as far as i understood, instead we could have STATUS_WRITE perm for the current perm rather than just WRITE... but fine if you think its an overkill

Reply:
for some reason they all used service names for the resource types. rather than being more specific writes/reads, one service may have multiple resource types with in them or i might have misunderstood something there, but thats fine let not worry too much

Reply:
yes there's a way to have a more granular level of permissions, that can be done like how SDES does here: https://github.com/hmrc/internal-auth-config/blob/main/qa/grants.yaml#L53-L82

Reply:
we can add as many permissions as we want, we just need to raise another PR with platops.

Reply:
but I see what you mean, in effect I didn't include the permission action in the name/description because the other teams didn't follow that format, but in retrospect, nothing is stopping us from doing that.

Reply:
cool, i am ok if you want to keep it simple for now, this will only be a problem if going forward if we want to add more endpoints with more perms

Reply:
we can add more `resourceLocations` with different resource names, and remove the all-encompassing asterisk - I don't think it's a problem for now as gvms-haulier-status is a rather simple service.

Reply:
if its easy to add go for it otherwise no issues

Reply:
unfortunately, that requires raising another SUP ticket with PlatOps, then for them to merge the PR in internal-auth.